cmd/readiness/main.go

```diff
@@ -89,6 +89,11 @@ func isPodReady(conf config.Config) (bool, error) {
 func hasDeadlockedSteps(health health.Status) bool {
 	currentStep := findCurrentStep(health.ProcessPlans)
 	if currentStep != nil {
+		// this means we are waiting for external factors before we are able to continue our update. If we expect the node to be up, then we can assume that it is ready.
+		// To match the rest of the code, a more proper way would be to look at all previous steps and
+		if currentStep.Step == "WaitCanUpdate" && currentStep.Started != nil && currentStep.Completed == nil && currentStep.Result == "wait" {
```

Comment:
alternative: adding `WaitCanUpdate` to parts of the deadlock risky steps

Reply:
It makes sense to me to move this to the risky steps. Is it a check that we might want to handle differently in the future and it's worth keeping separate?

Reply:
Done in 3c1576a !


The only "con"/difference I see is the 15 second health check: https://github.com/mongodb/mongodb-kubernetes-operator/blob/3c1576a6c6987f1ae5e07493f0fa58862fd80f79/cmd/readiness/main.go#L145

which means that our readinessProbe will "fail" for a max of 10 seconds until the time has gone. Which should not impact the version change process as after those 10 seconds we will return "ready" probes